### PR TITLE
[TextFields] Add baseline constraint support

### PR DESCRIPTION
**Pull request by willlarche**
Closes b/72401468

### CODE REVIEW

File: components/TextFields/src/MDCTextField.m [hunk 1]
```diff
@@ -226,6 +228,25 @@ - (void)updateBorder {
   self.borderView.borderPath = self.borderPath;
 }
 
+#pragma mark - Input Layout Strut Implementation
+
+- (void)setupInputLayoutStrut {
+  self.inputLayoutStrut = [[UILabel alloc] initWithFrame:CGRectZero];
+  //self.inputLayoutStrut.alpha = 0.f;
```
Comment (ianegordon) [inline components/TextFields/src/MDCTextField.m]: nit: This line is duplicated below.  Remove?

File: components/TextFields/src/MDCTextField.m [hunk 2]
```diff
@@ -226,6 +228,25 @@ - (void)updateBorder {
...
+  [self addSubview:self.inputLayoutStrut];
+
+  self.inputLayoutStrut.numberOfLines = 1;
+  self.inputLayoutStrut.alpha = 0.f;
```
Comment (ianegordon) [inline components/TextFields/src/MDCTextField.m]: Do we want to set:
self.inputLayoutStruct.hidden = YES

File: components/TextFields/src/MDCTextField.h [hunk 3]
```diff
@@ -30,6 +30,17 @@ extern NSString *_Nonnull const MDCTextFieldTextDidSetTextNotification;
...
+/**
+ This label should always have the same layout as the input field (which is private API.)
+
+ Unfortunately the included private baseline strut (which is the label returned for baseline-based
+ auto layout) has bugs that keep it from matching custom layout. We recreate it but also allow it to
+ have a width in case someone needs other kinds of auto layout constraints based off the input.
+
+ It always has an alpha of 0.0.
+ */
+@property(nonatomic, nullable, strong) UILabel *inputLayoutStrut;
```
Comment (ianegordon) [inline components/TextFields/src/MDCTextField.h]: Can we make this non-null and readonly?

File: components/TextFields/src/MDCTextField.m [hunk 4]
```diff
@@ -672,6 +694,19 @@ + (BOOL)requiresConstraintBasedLayout {
...
 
+- (UIView *)viewForFirstBaselineLayout {
+  return self.inputLayoutStrut;
+}
+
+- (UIView *)viewForLastBaselineLayout {
+  return self.inputLayoutStrut;
+}
+
+// TODO: (larche) Remove when we drop iOS 9 support
```
Comment (ianegordon) [inline components/TextFields/src/MDCTextField.m]: s/larche/#4390/

#4390 is the tracking bug for all the cleanup we can do post iOS 9.

**Review by ianegordon — CHANGES_REQUESTED** [review]

File: components/TextFields/src/MDCTextField.m (hunk 1 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextField.m]: Oops. That's from testing.

Comment (willlarche) [reply components/TextFields/src/MDCTextField.m]: Done.

File: components/TextFields/src/MDCTextField.m (hunk 2 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextField.m]: Yeah. This is one of those times where everything still works when hidden.

File: components/TextFields/src/MDCTextField.h (hunk 3 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextField.h]: Done.

File: components/TextFields/src/MDCTextField.m (hunk 2 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextField.m]: Done.

File: components/TextFields/src/MDCTextField.m (hunk 4 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextField.m]: Done.

**Comment by willlarche** [comment]
@ianegordon PTAL

**Review by ianegordon — APPROVED** [review]

**Pull request merged**